src/components/editor/property/InputListLOC.jsx

```diff
@@ -130,41 +136,123 @@ const InputListLOC = (props) => {
 
   let error
   let groupClasses = 'form-group'
+  const classes = ['modal', 'fade']
+  let display = 'none'
+  const modalId = `InputLookupModal-${props.property.key}`
+
   if (displayValidations && !_.isEmpty(validationErrors)) {
     groupClasses += ' has-error'
     error = validationErrors.join(',')
   }
 
+  if (props.show) {
+    classes.push('show')
+    display = 'block'
+  }
+
+  const handleClick = (event) => {
+    event.preventDefault()
+    dispatch(showModal(modalId))
+  }
+
+  const close = (event) => {
+    props.hideModal()
+    event.preventDefault()
+  }
+
+  // TODO: New styling to fit description in #2478
+  const lookupSelection = props.lookupValues.map((lookupValue) => {
+    return (
+      <div key={lookupValue.key} class="lookup-value">
+        <span key={lookupValue.key}>{lookupValue.label || lookupValue.literal}</span>
+        <button
+          onClick={() => props.removeValue(lookupValue.key)}
+          aria-label={`Remove ${lookupValue.label}`}
+          data-testid={`Remove ${lookupValue.label}`}
+          className="close rbt-close rbt-token-remove-button">
+        <span aria-hidden="true"><FontAwesomeIcon className="trash-icon" icon={faTrashAlt} /></span>
+        </button>
+        <a href={lookupValue.uri}>
+          <span aria-hidden="true"><FontAwesomeIcon className="globe-icon" icon={faGlobe} /></span>
+        </a>
+      </div>
+    )
+  })
+
+  const modal = (
```

Comment:
Is there a way to create a Lookup modal component that both `InputListLOC` and `InputLookup` could instead of defining a local modal for each of these components?"

Reply:
Looking into that now. Not sure as they are slightly different, but those differences may be inconsequential.